### PR TITLE
Update pusher.md

### PR DESCRIPTION
**Pull request by jlee2600hz**

### CODE REVIEW

File: applications/pusher/doc/pusher.md [hunk 1]
```diff
@@ -1,36 +1,161 @@
...
+sup kapps_controller restart_app fcm
+sup kapps_controller restart_app gcm
+sup kapps_controller restart_app pusher
+
+
+### Update
+
+Use the latest RPM for pusher and new pusher-role.cfg
+
+kazoo-application-pusher-4.3-77.pr6229.37773.el7.centos.x86_64.rpm
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: remove

File: applications/pusher/doc/pusher.md [hunk 2]
```diff
@@ -1,36 +1,161 @@
...
+
+[root@apps001 ~]# sup kapps_controller start_app pusher
+{ok,[fcm,gcm,base64url,gun,apns,pusher]}
+[root@apps001 ~]# sup kapps_controller running_apps
+[blackhole,callflow,cdr,conference,crossbar,fax,hangups,media_mgr,milliwatt,omnipresence,pivot,pusher,registrar,reorder,stepswitch,sysconf]
+
+pusher doc in system_config will be created but we will need to edit it later.
+
+
+3. Add Apple and google/firebase credentials into pusher doc using sup commands. The Apple cert needs to be in PEM (text) format and should be located in the server.
```
[the remark this answers is not in the document]
Comment (jlee2600hz) [reply applications/pusher/doc/pusher.md]: ok

File: applications/pusher/doc/pusher.md (hunk 1 above)
Comment (jlee2600hz) [reply applications/pusher/doc/pusher.md]: ok

File: applications/pusher/doc/pusher.md [hunk 3]
```diff
@@ -1,36 +1,67 @@
...
+
+
+3. Add Apple and google/firebase credentials into pusher doc using sup commands. The Apple cert needs to be in PEM (text, no encryption) format and should be located in the server.
+
+sup pusher_maintenance add_apple_app  [app-id] /etc/kamailio/cert/[voip_qa].pem api.development.push.apple.com
+sup pusher_maintenance add_apple_app [app-id] /etc/kazoo/kamailio/certs/[voip_pushcert].pem
+
+sup pusher_maintenance add_google_app [app-id] [firebase token]
+
+
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: remove this

File: applications/pusher/doc/pusher.md [hunk 4]
```diff
@@ -1,36 +1,67 @@
...
-If the device is already registered, and the client is alive, Kamailio will allow the SIP transaction to continue and the call will be handled as usual
+Sample Contact header in REGISTER message
+
+Contact: <sip:user_1234567@192.168.1.1:61767;transport=TCP;ob>;reg-id=1;+sip.instance="<urn:uuid:00000000-0000-0000-0000-00004e05965e>";app-id=com.app.mymobileapp;pn-tok=34fretrwgtrvtrebyenbytrwvtrebytenytbyrtebytebytebyten yhtevtrw;pn-type=apple
 
 ## Configuration
 
 ### System Config
 
-* `User-Agents`: list of user agents to check for pusher properties.
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: don't remove this

File: applications/pusher/doc/pusher.md [hunk 5]
```diff
@@ -1,36 +1,67 @@
...
-   }
-```
 
-### Maintenance
 
-In order for the push services from Apple / Firebase to work they need to be configured with application secrets / certificates. The app used in the push message is taken from Token-App.
 
-* `sup pusher_maintenance add_firebase_app(AppId, Secret)`
-* `sup pusher_maintenance add_apple_app(AppId, CertFile)` (uses the default APNs host: api.push.apple.com)
-* `sup pusher_maintenance add_apple_app(AppId, CertFile, Host)` (uses a custom APNs host, i.e. api.development.push.apple.com)
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: don't remove this

File: applications/pusher/doc/pusher.md [hunk 6]
```diff
@@ -34,3 +63,30 @@ In order for the push services from Apple / Firebase to work they need to be con
...
+2. Initialize pusher app.
+
+[root@apps001 ~]# sup kapps_controller start_app pusher
+{ok,[fcm,gcm,base64url,gun,apns,pusher]}
+[root@apps001 ~]# sup kapps_controller running_apps
+[blackhole,callflow,cdr,conference,crossbar,fax,hangups,media_mgr,milliwatt,omnipresence,pivot,pusher,registrar,reorder,stepswitch,sysconf]
+
+pusher doc in system_config will be created.
+
+
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: remove this

File: applications/pusher/doc/pusher.md [hunk 7]
```diff
@@ -34,3 +63,30 @@ In order for the push services from Apple / Firebase to work they need to be con
...
+3. Add Apple and google/firebase credentials into pusher doc using sup commands. The Apple cert needs to be in PEM (text, no encryption) format and should be located in the server.
+
+sup pusher_maintenance add_apple_app  [app-id] /etc/kamailio/cert/[voip_qa].pem api.development.push.apple.com
+sup pusher_maintenance add_apple_app [app-id] /etc/kazoo/kamailio/certs/[voip_pushcert].pem
+
+sup pusher_maintenance add_google_app [app-id] [firebase token]
+
+
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: remove

File: applications/pusher/doc/pusher.md [hunk 8]
```diff
@@ -34,3 +63,30 @@ In order for the push services from Apple / Firebase to work they need to be con
...
+[root@apps001 ~]# sup kapps_controller start_app pusher
+{ok,[fcm,gcm,base64url,gun,apns,pusher]}
+[root@apps001 ~]# sup kapps_controller running_apps
+[blackhole,callflow,cdr,conference,crossbar,fax,hangups,media_mgr,milliwatt,omnipresence,pivot,pusher,registrar,reorder,stepswitch,sysconf]
+
+pusher doc in system_config will be created.
+
+
+3. Add Apple and google/firebase credentials into pusher doc using sup commands. The Apple cert needs to be in PEM (text, no encryption) format and should be located in the server.
```
Comment (lazedo) [inline applications/pusher/doc/pusher.md]: move this (remove  number)  and combine with what is already mentioned in the 1st paragraph of `maintenance`

**Pull request merged**